src/rpc/net.cpp

```diff
@@ -1014,52 +1015,107 @@ static RPCHelpMan sendmsgtopeer()
+        "This RPC is for testing only.\n",
+        {
+            {"verbose", RPCArg::Type::BOOL, RPCArg::Default{false}, "True for a list of address manager entries by table, false for address counts only"},
+        },
+        {
+            RPCResult{"for verbose = false", RPCResult::Type::OBJ_DYN, "", "json object with network type as keys",
+                {
+                    {RPCResult::Type::OBJ, "network", "the network (" + Join(GetNetworkNames(), ", ") + ", all_networks )", {
+                        {RPCResult::Type::NUM, "new", "number of addresses in the new table, which represent potential peers the node has discovered but hasn't yet successfully connected to."},
+                        {RPCResult::Type::NUM, "tried", "number of addresses in the tried table, which represent peers the node has successfully connected to in the past."},
+                        {RPCResult::Type::NUM, "total", "total number of addresses in both new/tried tables"},
+                    }},
+                },
+            },
+            RPCResult{"for verbose = true", RPCResult::Type::OBJ_DYN, "", "json object with network type, new_table and tried_table as keys",
+                {
+                    {RPCResult::Type::ELISION, "", "The same output as verbose = false"},
+                    {
+                        RPCResult::Type::ARR, "table", "list of addresses in the address manager table ( new_table, tried_table )",
+                        {
+                            {RPCResult::Type::OBJ, "", "an address manager table entry", {
+                                {RPCResult::Type::STR, "address", "the address"},
+                                {RPCResult::Type::NUM, "services", "the services the node might support"},
+                                {RPCResult::Type::NUM, "bucket", "the address manager bucket the address is placed in"},
+                                {RPCResult::Type::NUM, "position", "the bucket position the address is placed in"},
+                                {RPCResult::Type::STR, "source", "the address that relayed the address to us"},
+                            }},
+                        },
+                    },
+                },
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("getaddrmaninfo", "")
+            + HelpExampleCli("getaddrmaninfo", "true")
+            + HelpExampleRpc("getaddrmaninfo", "")
+            + HelpExampleRpc("getaddrmaninfo", "true")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            NodeContext& node = EnsureAnyNodeContext(request.context);
+            if (!node.addrman) {
+                throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Address manager functionality missing or disabled");
+            }
+
+            bool verbose = false;
+            if (!request.params[0].isNull()) {
+                verbose = request.params[0].get_bool();
+            }
```

Comment:
Add

```c++
TMPL_INST(CheckRequiredOrDefault, bool, CHECK_NONFATAL(maybe_arg)->get_bool(););
```

in rpc/util.cpp and this becomes `const bool verbose = self.Arg<bool>(0);`  (see #28230)

Reply:
I choose to move to `getrawaddrman` instead of extending `getaddrmaninfo` with a verbose flag. https://github.com/bitcoin/bitcoin/pull/28523#discussion_r1339454178. Resolving this as there's no need for it anymore.